packages/@lwc/synthetic-shadow/src/faux-shadow/portal.ts

```diff
@@ -40,35 +43,23 @@ function adoptChildNode(node: Node, fn: ShadowRootResolver, shadowToken: string
     }
 }
 
-function disownChildNode(node: Node) {
-    // Note: this will remove the node owner key.
-    setShadowRootResolver(node, function() {} as ShadowRootResolver);
-
-    if (node instanceof Element) {
-        setShadowToken(node, undefined);
-        const childNodes = getInternalChildNodes(node);
-        for (let i = 0, len = childNodes.length; i < len; i += 1) {
-            const child = childNodes[i];
-            disownChildNode(child);
-        }
-    }
-}
-
 function initPortalObserver() {
     return new MO(mutations => {
         forEach.call(mutations, mutation => {
             const { target: elm, addedNodes, removedNodes } = mutation;
             // the target of the mutation should always have a ShadowRootResolver attached to it
             const fn = getShadowRootResolver(elm) as ShadowRootResolver;
             const shadowToken = getShadowToken(elm);
+
+            // First lets process removals in case we are in a case of removed and reinserted elements
+            for (let i = 0, len = removedNodes.length; i < len; i += 1) {
```

Comment:
this still have the problem of inserting and removing on the same tick :), only way to deal with this is to check if the removed element is connected :(, I don't think perf is an issue here... but it will be safer.

Reply:
i dont quite follow, i test this (see the added test). i noticed that the observer gets called 2 times, 1 with the node being added and another with the node being removed, but in the same order, so the last is the one that wins.

So you are worried about performance right? that we shouldn't add the key when being added if we know is going to be removed (which btw, is a penalty that we pay today in our master branch)

Reply:
@caridy is [this](https://github.com/salesforce/lwc/pull/1590/commits/fb9262db380a970908e68d12453d4018025cae2e) what you mean by that check?